neo/api/utils.py

```diff
@@ -1,24 +1,46 @@
 import json
+import gzip
 from functools import wraps
 
+COMPRESS_FASTEST = 1
+BASE_STRING_SIZE = 49
+MTU_TCP_PACKET_SIZE = 1500
+COMPRESS_THRESHOLD = MTU_TCP_PACKET_SIZE + BASE_STRING_SIZE
+
 
 # @json_response decorator for class methods
 def json_response(func):
     """ @json_response decorator adds header and dumps response object """
+
     @wraps(func)
     def wrapper(self, request, *args, **kwargs):
         res = func(self, request, *args, **kwargs)
+        response_data = json.dumps(res) if isinstance(res, dict) else res
         request.setHeader('Content-Type', 'application/json')
-        return json.dumps(res) if isinstance(res, dict) else res
+
+        if len(response_data) > COMPRESS_THRESHOLD:
+            accept_encoding = request.getHeader('Accept-Encoding')
```

Comment:
The idea here is to only apply compression when the data is actually larger than a single network packet, otherwise we're just wasting CPU cycles compressing data while it still goes into the same packet.
An empty string is 49 bytes in size, every additional character is 1 byte. So we do a rough estimate of the threshold to use.